### ISSUE TITLE
All extension tag is invalid in ebml_matroska.xml

### ISSUE THREAD

**Issue by nerg4l** [ietf-wg-cellar/matroska-specification#454]
ietf-wg-cellar/ebml-specification#270 defined `extension` with a required `type` attribute. Because of this missing attribute `ebml_matroska.xml` marked invalid when validated.

In the comments of that PR @dericed said the following (https://github.com/ietf-wg-cellar/ebml-specification/pull/270#issuecomment-515575870):

> [...] Once this is merged, I can adjust the corresponding pr at matroska.

I think #325 was "the corresponding pr" which I'm afraid was not adjusted.

**Comment by robUx4** [comment]
You're right it doesn't match the text of the EBML RFC. The validation of the Matroska spec is done via the XSD file from the EBML repository. That file is also incorrect, it doesn't mention the `type` attribute of the `extension` element, and even less that it's required.

You may file another Pull Request similar to https://github.com/ietf-wg-cellar/ebml-specification/pull/396 for this error in the XSD file. Then, once it's merged the Matroska repo will not validate anymore.

**Comment by nerg4l** [comment]
Maybe I'm wrong, but `extensionType` is defined the following way:

```
  <xs:complexType name="extensionType">
    <xs:sequence>
      <xs:any processContents="skip"
        minOccurs="0" maxOccurs="unbounded"/>
    </xs:sequence>
    <xs:attribute name="type" use="required"/>
    <xs:anyAttribute processContents="skip"/>
  </xs:complexType>
```

There for, it is defined correctly. I created a small Java app to validate the XML with the XSD and the validation failed. Also, JetBrains IDEs mark all extension tag invalid with a message which says `type` attribute is required. How is the XML validated? (I did not had a close look on that.)

**Comment by robUx4** [comment]
Ah yes, for some reason I thought it was not properly tied to `<xs:element name="extension"` but it is. Not sure why `xmllint` we use  here is not catching the missing required attribute...
